src/adapters/rubicon.js

```diff
@@ -107,70 +107,68 @@ var RubiconAdapter = function RubiconAdapter() {
   }
 
   /**
-   * Create (successful) bids for a unit,
+   * Create (successful) bids for a slot,
    * based on the given response
-   * @param {String} placement placement code/unit path
-   * @param {Object} response the response from rubicon
-   * @return {Bid} a bid objectj
+   * @param {Object} slot the slot from rubicon
+   * @param {Object} ads the raw responses
    */
-  function _makeBids(response, ads) {
+  function _makeBids(slot, ads) {
 
-    // if there are multiple ads, sort by CPM
-    ads = ads.sort(_adCpmSort);
-
-    var bidResponses = [];
-
-    ads.forEach(function(ad) {
-
-      var bidResponse,
-          size = ad.dimensions;
-
-      if (!size) {
-        // this really shouldn't happen
-        utils.logError('no dimensions given', RUBICON_BIDDER_CODE, ad);
-        bidResponse = _errorBid(response, ads);
-      } else {
-        bidResponse = bidfactory.createBid(1, response.bid);
+    if (!ads || ads.length === 0) {
 
-        bidResponse.bidderCode = RUBICON_BIDDER_CODE;
-        bidResponse.cpm = ad.cpm;
+      bidmanager.addBidResponse(
+        slot.getElementId(),
+        _errorBid(slot, ads)
+      );
 
-        // the element id is what the iframe will use to render
-        // itself using the rubicontag.renderCreative API
-        bidResponse.ad = _creative(response.getElementId(), size);
-        bidResponse.width = size[0];
-        bidResponse.height = size[1];
+    } else {
 
-        // DealId
-        if (ad.deal) {
-          bidResponse.dealId = ad.deal;
-        }
-      }
+      // if there are multiple ads, sort by CPM
+      ads = ads.sort(_adCpmSort);
 
-      bidResponses.push(bidResponse);
+      ads.forEach(function (ad) {
+        _makeBid(slot, ad);
+      });
 
-    });
+    }
 
-    return bidResponses;
   }
 
   /**
-   * Add success/error bids based
-   * on the response from rubicon
-   * @param {Object} response -- AJAX response from fastlane
+   * Create (successful) bid for a slot size,
+   * based on the given response
+   * @param {Object} slot the slot from rubicon
+   * @param {Object} ad a raw response
    */
-  function _addBids(response, ads) {
-    // get the bid for the placement code
-    var bids;
-    if (!ads || ads.length === 0) {
-      bids = [ _errorBid(response, ads) ];
+  function _makeBid(slot, ad) {
```

Comment:
this function just came from pulling singular functionality out of `_makeBids` which I didn't name and still exists.  I could rename them both, but for now I think I'll just leave it.